google/cloud/storage/internal/async/object_descriptor_impl.cc

```diff
@@ -200,7 +224,7 @@ void ObjectDescriptorImpl::Resume(google::rpc::Status const& proto_status) {
 void ObjectDescriptorImpl::OnResume(StatusOr<OpenStreamResult> result) {
   if (!result) return OnFinish(std::move(result).status());
   std::unique_lock<std::mutex> lk(mu_);
-  stream_ = std::move(result->stream);
+  streams_[0] = std::move(result->stream);
```

Comment:
I know nothing about this code, but it looks a little weird that this isn't assigning the "active stream".